core/account-id/src/lib.rs

```diff
@@ -55,9 +55,10 @@ impl AccountId {
                 last_char_is_separator = current_char_is_separator;
             }
 
-            (!last_char_is_separator)
-                .then(|| ())
-                .ok_or_else(|| ParseAccountError(ParseErrorKind::Invalid, account_id.to_string()))
+            if last_char_is_separator {
```

Comment:
Out of curioucity, does any test fail if we change this to `if !last_char_is_separator`

Reply:
@matklad I'll test it. I excluded this change from this PR, as this PR is supposed to be about `near-performance-metrics` and this change seems out of scope.